js/controller.js

```diff
@@ -598,8 +598,8 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
         this.state.screenToShow = CONSTANTS.SCREEN.END_SCREEN;
         this.mb.publish(OO.EVENTS.END_SCREEN_SHOWN);
       }
-      if (Utils.isIPhone()){
-        //iPhone end screen is the same as start screen, except for the replay button
+      if (!Utils.canRenderSkin()){
```

Comment:
I'm leaning towards having the components decide what to render. This way, the screenToShow states will be consistent with desktop flow (which hopefully means less headaches for this feature), but the components can just choose to not render anything. @pilievOoyala @aldigun @jpfranco your thoughts on this?

Reply:
I with you on this for the most part, unless we need to show a completely different screen because it doesn't make sense on that platform.

Reply:
I agree that the checks are ugly, but it looks like there is a bit more logic than just the rendering, so it might be tricky to migrate it to the components. We should definitely look for a way to refactor this, but for now I think it's safer to leave it as is.